src/main/java/org/folio/processing/mapping/mapper/writer/common/JsonBasedWriter.java

```diff
@@ -59,6 +67,91 @@ protected void writeObjectValue(String fieldPath, MapValue mapValue) {
     setValueByFieldPath(fieldPath, objectNode);
   }
 
+  @Override
+  protected void writeRepeatableValue(String fieldPath, RepeatableFieldValue value) {
+    List<Map<String, Value>> repeatableFields = value.getValue();
+    MappingRule.RepeatableFieldAction action = value.getRepeatableFieldAction();
+    for (Map<String, Value> subfield : repeatableFields) {
+      JsonNode currentObject = objectMapper.createObjectNode();
+      for (Map.Entry<String, Value> objectFields : subfield.entrySet()) {
+        JsonNode field = MissingNode.getInstance();
+        switch (objectFields.getValue().getType()) {
+          case LIST:
+          case MAP:
+            field = objectMapper.valueToTree(objectFields.getValue().getValue());
+            break;
+          case STRING:
+            field = new TextNode((String) objectFields.getValue().getValue());
+            break;
+          case BOOLEAN:
+            BooleanValue bool = (BooleanValue) objectFields.getValue();
+            MappingRule.BooleanFieldAction booleanFieldAction = bool.getValue();
+            if (booleanFieldAction.equals(MappingRule.BooleanFieldAction.ALL_TRUE)) {
+              field = BooleanNode.TRUE;
+            } else if (booleanFieldAction.equals(MappingRule.BooleanFieldAction.ALL_FALSE)) {
+              field = BooleanNode.FALSE;
+            }
+            break;
+          case MISSING:
+          case REPEATABLE:
+          default:
+            break;
+        }
+        if (!field.isMissingNode()) {
+          setValueByFieldPath(objectFields.getKey().substring(objectFields.getKey().indexOf(".") + 1), field, currentObject);
+        }
+      }
```

Comment:
I think it would be better to extract this code to the method. Moreover, there will be only 2 arguments!